File: java/dubbo-demo/dubbo-demo-api/src/main/java/com/alibaba/dubbo/demo/UserRequest.java
package com.alibaba.dubbo.demo;

import java.io.Serializable;

/**
 * Created by master on 21/11/2017.
 */
public class UserRequest implements Serializable {
    private Integer id;
    private String name;
    private String email;


    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Override
    public String toString() {
        return "UserRequest{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                '}';
    }
}

File: examples/hello-egg/app/dubbo/providers/com/alibaba/dubbo/demo/UserRequest.ts
import {Sex} from './Sex';
import java from 'js-to-java';

export interface IUserRequest {
  sex?: Sex;
  name?: string;
  id?: number;
  email?: string;
}

export class UserRequest {
  constructor(params: IUserRequest) {
    this.sex = params.sex;
    this.name = params.name;
    this.id = params.id;
    this.email = params.email;
  }

  sex?: Sex;
  name?: string;
  id?: number;
  email?: string;

  __fields2java() {
    return {
      $class: 'com.alibaba.dubbo.demo.UserRequest',
      $: {
        sex: java['enum']('com.alibaba.dubbo.demo.Sex', Sex[this.sex]),
        name: java.String(this.name),
        id: java.Integer(this.id),
        email: java.String(this.email),
      },
    };
  }
}

//generate by interpret-cli dubbo2.js
